rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/rascal/RascalTextDocumentService.java

```diff
@@ -287,8 +289,12 @@ public CompletableFuture<WorkspaceEdit> rename(RenameParams params) {
             .thenApply(Versioned::get)
             .handle((t, r) -> (t == null ? (file.getMostRecentTree().get()) : t))
             .thenCompose(tr -> rascalServices.getRename(tr, params.getPosition(), workspaceFolders, facts::getPathConfig, params.getNewName(), columns).get())
-            .thenApply(c -> new WorkspaceEdit(DocumentChanges.translateDocumentChanges(this, c)))
-            ;
+            .thenApply(t -> {
+                WorkspaceEdit wsEdit = new WorkspaceEdit();
+                wsEdit.setDocumentChanges(DocumentChanges.translateDocumentChanges(this, (IList) t.get(0)));
+                wsEdit.setChangeAnnotations(DocumentChanges.translateChangeAnnotations((IMap) t.get(1)));
```

Comment:
1. Mark changes as possibly requiring user interaction before being applied.
2. Provide a description.

In the UI, that might look like this:
![afbeelding](https://github.com/user-attachments/assets/4a8133aa-2e19-4f27-9f7f-891114fa04b3)

Reply:
I was a bit surprised that the edits that we produced before - that were applied without user interaction - are now suddenly also shown in this list, and can be unconfirmed by the user. I'll experiment with this and see if we can prevent that. Otherwise, we can attach a description to them as well, to persuade the user not to touch them.

Reply:
If we send any edits with `needsConfirmation = true` (like right-hand sides of `has`), all edits will [always appear](https://github.com/microsoft/vscode/blob/6e3cbfd138073c018087744c5597209f61f0ad18/src/vs/workbench/contrib/bulkEdit/browser/bulkEditService.ts#L190) in this 'refactor preview'. I do not see any way to lock the non-confirmable edits to prevent them being unchecked by the user.

Therefore, I followed your suggestion and added some descriptions to the headers. What do you think of the following @DavyLandman ?

![afbeelding](https://github.com/user-attachments/assets/0258bf47-1880-4318-9590-efc1cc243598)

Reply:
If there are no to-be-confirmed instances (currently: no uses of `has <name-under-cursor>`), the preview will still be skipped.

Reply:
Yes, much better, how about rewording it:

> Due to the dynamic behavior of these uses, we are not sure if they should also be renamed, please review these changes.

and

> We're positive these edits are necessary, they are only shown here to get a full overview, we advice not to disable them.

Reply:
good enough :+1: